target/scripts/helpers/utils.sh

```diff
@@ -145,3 +145,39 @@ function _env_var_expect_integer() {
   _log 'warn' "The value of '${ENV_VAR_NAME}' is not an integer ('${!ENV_VAR_NAME}'), but was expected to be"
   return 1
 }
+
+# Ensures that zenv only runs envsubst with ENV filtered from the provided prefix.
+# Those ENV are loaded by zenv in the format of an `.env` file (with prefix dropped by sed)
+# When an ENV is not available, envsubst will evaluate it as empty.
+#
+# @param ${1} = Use a prefix for a group of environment variables
+# @param ${2} = Filepath to ENV template
+# @output     = Template file content populated with available ENV
+function _template_with_env() {
+  local ENV_PREFIX=${1:?ENV prefix is required}
+  local ENV_TEMPLATE=${2:?ENV template filepath is required}
+
+  if [[ ! -f ${ENV_TEMPLATE} ]]; then
+    _dms_panic__invalid_value "file '${ENV_TEMPLATE}' does not exist" 'utils.sh:_use_env_template'
+  fi
+
+  # NOTE: $PATH is retained to avoid needing absolute paths for binaries.
+  env --ignore-environment PATH="${PATH}" \
+    zenv --file <(env | grep "^${ENV_PREFIX}" | sed "s/^${ENV_PREFIX}//") \
+    envsubst < "${ENV_TEMPLATE}"
```

Comment:
That's kind of handled above the function name? Could I phrase that better or is it more of an issue with the location?

Reply:
I think the function description should give a high-level overview of what the function does, not how it is implemented. I'd move down the description and just add a single sentence about what the function does.

Reply:
```suggestion
  # Ensures that zenv only runs envsubst with ENV filtered from the provided prefix.
  # Those ENV are loaded by zenv in the format of an `.env` file (with prefix dropped by sed)
  # When an ENV is not available, envsubst will evaluate it as empty.
  #
  # NOTE: $PATH is retained to avoid needing absolute paths for binaries.
  env --ignore-environment PATH="${PATH}" \
    zenv --file <(env | grep "^${ENV_PREFIX}" | sed "s/^${ENV_PREFIX}//") \
    envsubst < "${ENV_TEMPLATE}"
```